rust/rope/src/rope.rs

```diff
@@ -490,7 +495,12 @@ impl Rope {
     }
 
     /// Returns a slice of the string from the byte range [`start`..`end`).
-    pub fn slice(&self, start: usize, end: usize) -> Rope {
+    pub fn slice<T>(&self, range: T) -> Rope 
+        where T: RangeBounds<usize>
+    {
+        let start = self.get_start(&range);
+        let end = self.get_end(&range);
```

Comment:
seeing this everywhere makes me wonder if we'd rather have a fn,

```rust
    self.extract_range(range: T) -> (usize, usize) { .. }
```

and then you could have,
```rust
let (start, end) = self.extract_range(&range);
```